### PR TITLE
Openstack keystone V3 API support in keystone

### PR DESCRIPTION
**Pull request by jirikotlin**
### What does this PR do?

Modified openstack keystone state and execution module to be both usable with new V3 API.
### Previous Behavior

Keystone V2 API support only.
### New Behavior

Supports both openstack keystone V2 and V3 API.
### Tests written?

No
### Previous PR:
#32017

### CODE REVIEW

**Comment by gtmanfred** [comment]
:+1: thanks!

File: salt/modules/keystone.py [hunk 1]
```diff
@@ -287,38 +323,44 @@ def endpoint_list(profile=None, **connection_args):
...
     for endpoint in kstone.endpoints.list():
-        ret[endpoint.id] = {'id': endpoint.id,
-                            'region': endpoint.region,
-                            'adminurl': endpoint.adminurl,
-                            'internalurl': endpoint.internalurl,
-                            'publicurl': endpoint.publicurl,
-                            'service_id': endpoint.service_id}
+        ret[endpoint.id] = dict((value, getattr(endpoint, value)) for value in dir(endpoint)
+                                if not value.startswith('_') and
+                                isinstance(getattr(endpoint, value), (unicode, dict, bool)))
```
Comment (cachedout) [inline salt/modules/keystone.py]: This is not Python 3 compatible. Please change all the references from `unicode` to `six.text_type`. Thanks.

**Comment by cachedout** [comment]
@jirikotlin A number of related tests seem to be failing. Could you please take a look? Thanks.

**Comment by gtmanfred** [comment]
@jirikotlin can you take a look at these https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/4220/ ?  We are going to be tagging Carbon soon, and I really want to make sure that this gets into carbon.

Thanks!
Daniel

**Comment by jirikotlin** [comment]
Hi, working on it but currently do not have much time, hope it will be this week.

**Comment by gtmanfred** [comment]
This pr should finish the last of the unittests for this pr.

https://github.com/saltstack/salt/pull/35911

**Pull request closed without merging**